kbs/src/attestation/backend.rs

```diff
@@ -187,9 +192,12 @@ impl AttestationService {
             );
         }
 
+        // For now just use the first TEE to get the challenge.
+        // Currently the only platform that requires its own challenge
+        // is s390x, which does not yet require attestation of devices.
         let challenge = self
             .inner
-            .generate_challenge(request.tee, request.extra_params.clone())
+            .generate_challenge(request.tees[0], request.extra_params.clone())
```

Comment:
What if the first TEE is a non-CPU one? With no report data?

Reply:
Yeah this should change to check if any of the TEEs are s390x.